### PR TITLE
CircleCI setup and testApp

### PR DESCRIPTION
**Pull request by PravinPK**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+checkstyle:
```
Comment (PravinPK) [inline Makefile:1]: I haven't prefixed `notificationbuilder-` to all the make command. Thinking we can wait until the next UIComponent is added to edit these command and the ci files.
Let me know

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      - run:
+          name: Build Test app
+          command: make assemble-app
+
+      - run:
+          name: Run Unit tests
+          command: make unit-test-coverage
+
+      - store_test_results:
+          path: code/userprofile/build/test-results/testPhoneDebugUnitTest
```
Comment (praveek) [inline .circleci/config.yml:61]: notificationbuilder

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+      - codecov/upload:
+          file: code/userprofile/build/reports/coverage/test/phone/debug/report.xml
```
Comment (praveek) [inline .circleci/config.yml:64]: notificationbuilder

File: .circleci/config.yml (hunk 2 above)
Comment (PravinPK) [reply .circleci/config.yml:61]: Copy-pasta error. good catch :P

**Review by praveek — APPROVED** [review]

**Pull request merged**